lib/filter_lib/src/pagination.py

```diff
@@ -9,24 +9,48 @@
 T = TypeVar("T")
 PaginationParams = namedtuple(
     "PaginationParams",
-    ["page_num", "page_size", "min_pages_left", "total", "has_more"],
+    [
+        "page_num",
+        "page_size",
+        "min_pages_left",
+        "total",
+        "has_more",
+        "page_offset",
+    ],
 )
 
 
 def make_pagination(
-    query: Query, page_number: int, page_size: int
+    query: Query, page_number: int, page_size: int, page_offset: int
 ) -> Tuple[Query, PaginationParams]:
-    has_more: bool = False
-    max_count = page_size * 10 + 1
-    total_results: int = query.limit(max_count).count()
-    if total_results == max_count:
-        has_more = True
-        total_results -= 1
-    query = query.limit(page_size)
-    query = query.offset((page_number - 1) * page_size)
+    if page_number is not None:
```

Comment:
Please, simplify this code, you have got 2 branches with similar algorithm

Reply:
The logic in those two branches differ.

When using 'page_number' (old way):
offset = (page_number - 1) * page_size
'total_results' is calculated for 10 pages only.
'has_more' shows whether there are more entries that are in first 10 pages.

When using 'page_offset' (new_functionality):
offset = page_offset passed in (page_num is not used in calculation as above)
'total_results' is calculated for all the results that satisfy query filters - if the logic should be different, please write down what should 'total_results' mean in this particular case
'has_more' shows whether there are more entries after the "limit" == "page_size". No logic about 10 pages